ehitchhiking-ui/src/app/shared/components/active-trips-modal/active-trips-modal.component.ts

```diff
@@ -48,15 +47,46 @@ export class ActiveTripsModalComponent implements OnInit {
 
 	ngOnInit() {
 		this.scrollObserver.observe(this.markerRef.nativeElement);
-		this.tripsArrayLenght = this.tripService.getTrips().length;
 		this.fetchTrips();
 	}
 
 	fetchTrips() {
-		this.tripsArray = this.tripService.getTrips();
-		setTimeout(() => {
-			this.loading = false;
-		}, 1000);
+		this.loading = true;
+		this.activeTripsApiService
+			.getActiveTrips()
+			.then((data) => {
+				this.tripsArray = data;
+			})
+			.finally(() => {
+				this.loading = false;
+			});
+	}
+
+	removeTripPassenger(id: number) {
+		this.activeTripsApiService
+			.removeTripPassenger(id)
+			.then(() => {
+				this.fetchTrips();
+			})
+			.catch(() => {
+				this.loading = false;
+			});
+	}
+
+	removeTripDriver(id: number) {
+		this.activeTripsApiService
+			.removeTripDriver(id)
+			.then(() => {
+				this.fetchTrips();
+			})
+			.catch(() => {
+				this.loading = false;
+			});
+	}
+
+	onDelete($event) {
+		this.loading = true;
```

Comment:
const request = $event.role === UserState.Passenger ? this.removeTripPassenger : this.removeTripDriver;
request($event.id);

Reply:
okay

Reply:
`if ($event.role === UserState.Passenger) {
    this.removeTripPassenger($event.id);
    return;
}

this.removeTripDriver($event.id);
`